message_ix/model/MESSAGE-MACRO_run.gms

```diff
@@ -257,18 +263,59 @@ if ( abs(max_adjustment) < %CONVERGENCE_CRITERION%,
     else
         put_utility 'log' /"+++ Convergence criteria satisfied after ", ORD(iteration):0:0, " iterations +++ " ;
     ) ;
+    if ( osc_check_final = 1,
+        put_utility 'log' /"+++ Convergence achieved via oscillation check mechanism; check iteration log for further details +++ " ;
```

Comment:
could we add here a log message based on the value of `osc_check` saying which method was used?

Reply:
This is in fact done. the log file will write for each of the iterations which of the checks has been activated, hence when convergence is achieved no specific "check" is referenced as multiple checks may have been active over the iterations.

Reply:
ah, nice